services/index/internal/controller/rest/node_handler.go

```diff
@@ -791,40 +757,44 @@ func handleAddNodeErrors(c *gin.Context, err error) {
 	c.JSON(jsonErr[0].Status, res)
 }
 
-func handleGetNodeErrors(c *gin.Context, err error, nodeID string) {
+func handleGetNodeErrors(c *gin.Context, err error, nodeID *string) {
 	var notFoundError index.NotFoundError
 	var databaseError index.DatabaseError
 	var jsonErr []jsonapi.Error
 
-	switch {
-	case errors.As(err, &notFoundError):
+	if errors.As(err, &notFoundError) {
+		nodeIDMsg := "a node"
+		if nodeID != nil {
+			nodeIDMsg = fmt.Sprintf(
+				"the following node_id in the Index: %s",
+				*nodeID,
+			)
+		}
 		jsonErr = jsonapi.NewError(
 			[]string{"Node Not Found"},
-			[]string{
-				fmt.Sprintf(
-					"Could not locate the following node_id in the Index: %s",
-					nodeID,
-				),
-			},
+			[]string{fmt.Sprintf("Could not locate %s", nodeIDMsg)},
 			nil,
 			[]int{http.StatusNotFound},
 		)
+	} else {
+		logger.Error("Failed to get a node", err)
 
-	case errors.As(err, &databaseError):
-		jsonErr = jsonapi.NewError(
-			[]string{databaseError.Message},
-			[]string{"Error while trying to access the database."},
-			nil,
-			[]int{http.StatusInternalServerError},
-		)
-
-	default:
-		jsonErr = jsonapi.NewError(
-			[]string{"Unknown Error"},
-			[]string{"An unexpected error occurred. Please try again later."},
-			nil,
-			[]int{http.StatusInternalServerError},
-		)
+		switch {
+		case errors.As(err, &databaseError):
+			jsonErr = jsonapi.NewError(
+				[]string{databaseError.Message},
+				[]string{"Error while trying to delete a node."},
+				nil,
+				[]int{http.StatusNotFound},
+			)
+		default:
+			jsonErr = jsonapi.NewError(
+				[]string{"Unknown Error"},
+				[]string{},
```

Comment:
Did you miss the message here?
"[]string{"An unexpected error occurred. Please try again later."},"